tests/lib/rules/no-context.js

```diff
@@ -36,12 +36,59 @@ ruleTester.run('no-context', rule, {
     parserOptions: parserOptions
   }, {
     code: [
+      'var Hello = React.createClass({',
+      '  propTypes: {',
+      '    name: React.PropTypes.string',
+      '  },',
+      '  render: function() {',
+      '    return <div>Hello {this.props.name}</div>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions
+  }, {
+    code: [
       'var Hello = function() {',
       '  var context;',
       '  return context;',
       '};'
     ].join('\n'),
     parserOptions: parserOptions
+  }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  static propTypes = {',
+      '    name: React.PropTypes.string',
+      '  };',
+      '  render() {',
+      '    return <div>Hello {this.props.name}</div>;',
+      '  }',
+      '};'
+    ].join('\n'),
+    parser: 'babel-eslint',
+    parserOptions: parserOptions
+  }, {
+    code: [
+      'class Hello extends React.Component {',
```

Comment:
Might as well. Note: that will likely require a rebase onto origin/master (which you probably want to do anyway).